pkg/data/generic_test.go

```diff
@@ -474,9 +473,9 @@ func TestGenericSync(t *testing.T) {
 
```

Comment:
❓ How do you judge our test coverage here, does it inspire confidence? (I really don't know, it's a genuine question since you're working on this.)

Reply:
After #135 we have the initial load covered, and I'm confident on that part. But I have failed to add tests for addition / update / removal of resources after the initial load, and that's a problem.

I have tried both [FakeDynamicClient.Tracker](https://pkg.go.dev/k8s.io/client-go/dynamic/fake#FakeDynamicClient.Tracker) and [FakeDynamicClient.Resource(){.Create, .Update, .Delete}](https://github.com/kubernetes/client-go/blob/b425fb657986de75a11b1844886c70b486728935/dynamic/fake/simple.go#L155), but I couldn't make them work. The tests didn't work even with the previous version.

So I'm currently testing this "on the field", in my company's dev environment, before we deploy the service to our production. In parallel, I'm checking out the `FakeDynamicClient` code to try and find out why the tests don't work.

Reply:
I could make the tests work in 602be1fb11f5dd9ce0f912a69af1d49b8d39cab4 and now the coverage is around 78%. I think it's safe to remove the draft label now.